bin/query-cloudwatch

```diff
@@ -33,41 +34,163 @@ class QueryCloudwatch
     :num_threads,
     :wait_duration,
     :count_distinct,
+    :sqlite_database_file,
     keyword_init: true,
   )
 
+  class JsonOutput
+    attr_reader :stdout
+    
+    def initialize(stdout:)
+      @stdout = stdout
+    end
+
+    def abort; end
+
+    def start; end
+    
+    def handle_row(e)
+      stdout.puts row.to_json
+    end
+    
+    def finish; end
+  end
+
+  class CsvOutput
+    attr_reader :stdout
+    
+    def initialize(stdout:)
+      @stdout = stdout
+    end
+
+    def abort; end
+
+    def start; end
+    
+    def handle_row(row)
+      stdout.puts e.values.to_csv
+    end
+    
+    def finish; end
+  end
+
+  class SqliteOutput
+    attr_reader :db, :filename, :stderr
+
+
+    def initialize(filename:, stderr:)
+      @filename = filename
+      @stderr = stderr
+    end
+
+    def abort
+      if db
+        db.rollback if db.transaction_active?
+        db.interrupt
+        db.close
+        @db = nil
+      end
+    end
+
+    def start
+      @db = SQLite3::Database.new(filename)
+      create_tables_if_needed
```

Comment:
does this handle like appending to an existing file? or is it always all-new?

Reply:
It will create a database file if one doesn't exist. If one exists, it will insert rows into the `events` table, but it assumes each row has a unique id. If there is a collision on id, it will just not insert it.

Reply:
(I don't have a migration strategy right now because it seems like overkill)